crates/cast/src/main.rs

```diff
@@ -91,15 +88,15 @@ async fn main() -> Result<()> {
     let config = parse_scarb_config(&cli.profile, &cli.path_to_scarb_toml)?;
 
     let rpc_url = cli.rpc_url.unwrap_or(config.rpc_url);
-    let network = cli.network.unwrap_or(config.network);
     let account = cli.account.unwrap_or(config.account);
 
-    let provider = get_provider(&rpc_url, &network).await?;
+    let provider = get_provider(&rpc_url)?;
 
     match cli.command {
         Commands::Declare(declare) => {
             account_file_exists(&accounts_file_path)?;
-            let mut account = get_account(&account, &accounts_file_path, &provider, &network)?;
+            let chain_id = provider.chain_id().await?;
```

Comment:
Looks like chain_id is needed for each subcommand. Maybe we should move this aoutside match block?

Reply:
It's async, that's why I hesitated

Reply:
I mean, it's not used everywhere and it's async, that's why I hesitated*

Reply:
`multicall new` and `call` do not need it